modular_zubbers/code/datums/quirks/neutral_quirks/hoarder.dm

```diff
@@ -0,0 +1,253 @@
+	lose_text = span_warning("You've lost interest in your hoard.")
+	medical_record_text = "Patient has a compulsion to collect and hoard items."
+
+	var/list/hoard_turfs = list()
+	var/list/hoard_images = list()
+	var/hoard_visible = TRUE
+	var/hoard_process_interval = 30 SECONDS
+	var/hoard_grace_period = 10 MINUTES
+
+/datum/quirk/hoarder/add(client/client_source)
+	var/mob/living/carbon/human/human_holder = quirk_holder
+	var/datum/action/cooldown/spell/hoard/expand_hoard = new /datum/action/cooldown/spell/hoard()
+	expand_hoard.Grant(human_holder)
+	expand_hoard.hoard_quirk = src
+	var/datum/action/cooldown/spell/toggle_hoard/toggle_hoard = new /datum/action/cooldown/spell/toggle_hoard()
+	toggle_hoard.Grant(human_holder)
+	toggle_hoard.hoard_quirk = src
+	addtimer(CALLBACK(src, PROC_REF(first_hoard_check)), hoard_grace_period)
+
+/datum/quirk/hoarder/remove()
+	if(QDELETED(quirk_holder))
+		return ..()
+	var/datum/action/cooldown/spell/hoard/expand_hoard = locate(/datum/action/cooldown/spell/hoard) in quirk_holder.actions
+	expand_hoard.Remove()
+	var/datum/action/cooldown/spell/toggle_hoard/toggle_hoard = locate(/datum/action/cooldown/spell/hoard) in quirk_holder.actions
+	toggle_hoard.Remove()
+
+	// clean up overlays if any
+	if(quirk_holder?.client && hoard_images)
+		for(var/image/I in hoard_images)
+			if(I && quirk_holder.client)
+				quirk_holder.client.images -= I
+				qdel(I)
+		hoard_images = list()
+
+	return ..()
+
+
+// === Hoard Calculation Below ===
+/datum/quirk/hoarder/proc/first_hoard_check()
+	update_hoard_mood()
+	addtimer(CALLBACK(src, PROC_REF(update_hoard_mood)), hoard_process_interval, TIMER_LOOP)
+
+/datum/quirk/hoarder/proc/update_hoard_mood()
+	var/mob/living/carbon/human/H = quirk_holder
+	if(!H)
+		return
+	var/value = get_total_hoard_value()
+	if (value == 0)
+		H.add_mood_event("hoard_value", /datum/mood_event/hoard_none)
+		return
+	if (value < HOARD_BAD)
+		H.add_mood_event("hoard_value", /datum/mood_event/hoard_sparse)
+		return
+	if (value < HOARD_OKAY)
+		H.add_mood_event("hoard_value", /datum/mood_event/hoard_meager)
+		return
+	if (value < HOARD_GOOD)
+		H.add_mood_event("hoard_value", /datum/mood_event/hoard_acceptable)
+		return
+	if (value < HOARD_FULL)
+		H.add_mood_event("hoard_value", /datum/mood_event/hoard_plentiful)
+		return
+	H.add_mood_event("hoard_value", /datum/mood_event/hoard_overflowing)
+	return
+
+/datum/quirk/hoarder/proc/get_total_hoard_value()
+	if (!hoard_turfs)
+		return 0
+	var/final_value = 0
+	for (var/turf/T in hoard_turfs)
+		var/list/obj/item/hoard_items = T.contents
+		for(var/obj/item/trinket in hoard_items)
+			final_value += trinket.hoard_value
+	return final_value
+
+
+// === Hoard Expansion code below ===
+
+/datum/action/cooldown/spell/hoard
+	name = "Expand Hoard"
+	desc = "Make the tile you are standing on part of your hoard"
+
+	button_icon = 'icons/obj/devices/tool.dmi'
+	button_icon_state = "multitool"
+	cooldown_time = 0 SECONDS
```

Comment:
give it some kind of tiny cooldown to prevent spamming a turf lookup, 1 second is fine